### ISSUE TITLE
Remove rest_error_handler

### ISSUE THREAD

**Issue by Stoyan0v** [elementor/elementor#12219]
## Prerequisites

- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x] The issue still exists against the latest stable version of Elementor.

## Description

Dear Elementor Support, please remove `rest_error_handler` method, along with `add_action( 'rest_api_init', [ $this, 'register_rest_error_handler' ] );`, because these two methods are causing issues with every plugin, which uses WP REST API.

The problem is that when you set the following:
`set_error_handler( [ $logger_manager, 'rest_error_handler' ], E_ALL );`, you are allowing all errors/notices/warngins to be displayed. From there you show the error message if the WP_DEBUG is enabled or a message to see the System Info of Elementor.

It's not ok to show such messages when the debug is disabled. 
For example, your hook is used on our plugin page, and you show your message because our plugin generates the following notice: 
```
Deprecated: strpos(): Non-string needles will be interpreted as strings in the future
```

We know about this warning, and we will fix it soon, but I don't see a reason to show such message in OUR endpoint. Even when the notice is suppressed using `@`.

At least, if you want to have such error handler, check the error_file param or something else, but do not change the way, how WP debug works.

I will be happy to provide more information or anything else if you need my help.

Regards,
Stanimir

## Isolating the problem

- [ x ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ x ] This bug happens with a default WordPress theme active.

**Comment by shilo-ey** [comment]
Thanks for the detailed report.

We are aware of this issue and working on a fix.

We will update this thread when there's an update.

Thanks

**Comment by shilo-ey** [comment]
Thanks for the detailed report.

This issue has been resolved in Elementor v3.0.1

Feel free to update

Thanks!

**Comment by kukkakeppi** [comment]
3.0.1 works superbly here! Thank you @shilo-ey and team! 👍
